test/vegur_roundtrip_SUITE.erl

```diff
@@ -1003,6 +1003,31 @@ host_conflict(Config) ->
     {ok, Resp} = gen_tcp:recv(Client, 0, 1000),
     {match, _} = re:run(Resp, "400").
 
+drop_forwarded_host(Config) ->
+    %% Drop the X-Forwarded-Host headers because we never cared about them
+    %% before and some apps we proxy too can get nasty injection attacks from
+    %% them.
+    IP = ?config(server_ip, Config),
+    Port = ?config(proxy_port, Config),
+    Req =
+    "POST / HTTP/1.1\r\n"
+    "Host: "++domain(Config)++"\r\n"
+    "x-forwArded-Host: "++domain(Config)++"\r\n"
+    "Content-Type: text/plain\r\n"
+    "X-Forwarded-Host: "++domain(Config)++"\r\n"
+    "\r\n"
+    "0123456789",
+    %% Open the server to listening. We then need to send data for the
+    %% proxy to get the request and contact a back-end
+    Ref = make_ref(),
+    start_acceptor(Ref, Config),
+    {ok, Client} = gen_tcp:connect(IP, Port, [{active,false},list],1000),
+    ok = gen_tcp:send(Client, Req),
+    Server = get_accepted(Ref),
+    %% receive the request, content-length of 10 is there once only.
+    {ok, Proxied} = gen_tcp:recv(Server, 0, 1000),
+    nomatch = re:run(Proxied, "[xX]-[fF]orwarded-[hH]ost:", [{capture, all}, global]).
```

Comment:
Would it be better (/possible) to make this a case-insensitive match and verify `x-forwarded-host` does not appear at all? Then a change to the test setup wouldn't turn this into a false negative.